src/test/java/quickcache/storage/JsonQuickCacheStorageTest.java

```diff
@@ -1,110 +1,110 @@
-//        assertThrows(NullPointerException.class, () -> saveAddressBook(null, "SomeFile.json"));
-//    }
-//
-//    /**
-//     * Saves {@code addressBook} at the specified {@code filePath}.
-//     */
-//    private void saveAddressBook(ReadOnlyQuickCache addressBook, String filePath) {
-//        try {
-//            new JsonAddressBookStorage(Paths.get(filePath))
-//                    .saveAddressBook(addressBook, addToTestDataPathIfNotNull(filePath));
-//        } catch (IOException ioe) {
-//            throw new AssertionError("There should not be an error writing to the file.", ioe);
-//        }
-//    }
-//
-//    @Test
-//    public void saveAddressBook_nullFilePath_throwsNullPointerException() {
-//        assertThrows(NullPointerException.class, () -> saveAddressBook(new QuickCache(), null));
-//    }
-//}
+package quickcache.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static quickcache.testutil.Assert.assertThrows;
+import static quickcache.testutil.TypicalFlashcards.RANDOM1;
+import static quickcache.testutil.TypicalFlashcards.RANDOM2;
+import static quickcache.testutil.TypicalFlashcards.RANDOM3;
+import static quickcache.testutil.TypicalFlashcards.getTypicalQuickCacheForStorage;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import quickcache.commons.exceptions.DataConversionException;
+import quickcache.model.QuickCache;
+import quickcache.model.ReadOnlyQuickCache;
+
+public class JsonQuickCacheStorageTest {
+    private static final Path TEST_DATA_FOLDER = Paths.get("src", "test", "data", "JsonAddressBookStorageTest");
+
+    @TempDir
+    public Path testFolder;
+
+    @Test
+    public void readAddressBook_nullFilePath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> readAddressBook(null));
+    }
+
+    private java.util.Optional<ReadOnlyQuickCache> readAddressBook(String filePath) throws Exception {
+        return new JsonQuickCacheStorage(Paths.get(filePath)).readQuickCache(addToTestDataPathIfNotNull(filePath));
+    }
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    @Test
+    public void read_missingFile_emptyResult() throws Exception {
+        assertFalse(readAddressBook("NonExistentFile.json").isPresent());
+    }
+
+    @Test
+    public void read_notJsonFormat_exceptionThrown() {
+        assertThrows(DataConversionException.class, () -> readAddressBook("notJsonFormatAddressBook.json"));
+    }
+
+    @Test
+    public void readAddressBook_invalidPersonAddressBook_throwDataConversionException() {
```

Comment:
There are still instances of person and addressbook here and there, might need to rename them.

Reply:
I've made the changes, thanks!